File: app/src/main/java/com/arad/care4pets/RemindersActivity.java
package com.arad.care4pets;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageButton;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class RemindersActivity extends AppCompatActivity {

    private ReminderAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reminders);

        RecyclerView recycler = findViewById(R.id.rvReminders);
        FloatingActionButton fabAdd = findViewById(R.id.fabAddReminder);
        ImageButton btnBack = findViewById(R.id.btnBack);

        adapter = new ReminderAdapter(FakeRepository.reminders);
        recycler.setLayoutManager(new LinearLayoutManager(this));
        recycler.setAdapter(adapter);

        fabAdd.setOnClickListener(v ->
        startActivity(new Intent(RemindersActivity.this, AddReminderActivity.class))
        );

        btnBack.setOnClickListener(v -> finish());
    }

    @Override
    protected void onResume() {
        super.onResume();
        adapter.setReminders(FakeRepository.reminders);
    }
}

File: app/src/main/java/com/arad/care4pets/RemindersActivity.kt
package com.arad.care4pets

import android.content.Intent
import android.os.Bundle
import android.widget.ImageButton
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.floatingactionbutton.FloatingActionButton

class RemindersActivity : AppCompatActivity() {

    private lateinit var adapter: ReminderAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reminders)

        val recycler = findViewById<RecyclerView>(R.id.rvReminders)
        val fabAdd = findViewById<FloatingActionButton>(R.id.fabAddReminder)
        val btnBack = findViewById<ImageButton>(R.id.btnBack)

        adapter = ReminderAdapter(FakeRepository.reminders)
        recycler.layoutManager = LinearLayoutManager(this)
        recycler.adapter = adapter

        fabAdd.setOnClickListener {
            startActivity(Intent(this, AddReminderActivity::class.java))
        }

        btnBack.setOnClickListener {
            finish()
        }
    }

    override fun onResume() {
        super.onResume()
        adapter.reminders = FakeRepository.reminders
    }
}
